src/Agent/NewRelic/Profiler/Profiler/ICorProfilerCallbackBase.h

```diff
@@ -241,7 +266,10 @@ namespace Profiler {
             return eventMask;
         }
 
-        virtual bool ShouldInstrument(std::shared_ptr<Configuration::Configuration> configuration, xstring_t processPath, xstring_t commandLine, xstring_t appPoolId) = 0;
+        bool ShouldInstrument(std::shared_ptr<Configuration::Configuration> configuration, xstring_t processPath, xstring_t commandLine, xstring_t appPoolId)
+        {
+            return _isCoreClr ? configuration->ShouldInstrumentNetCore(processPath, appPoolId, commandLine) : configuration->ShouldInstrumentNetFramework(processPath, appPoolId);
```

Comment:
A potential further improvement could be to unify these two methods `ShouldInstrumentNetCore` and `ShouldInstrumentNetFramework` if possible, but it's not necessary now.